db/src/urls.rs

```diff
@@ -83,7 +85,7 @@ async fn download_raw_html(url: &str) -> color_eyre::Result<String> {
 fn clean_raw_html(raw_html: &str, url: &Url) -> color_eyre::Result<String> {
     let mut raw_html_cursor = std::io::Cursor::new(raw_html);
     let article = extractor::extract(&mut raw_html_cursor, url)?;
-    Ok(article.content)
+    Ok(article.text)
```

Comment:
in extractor function from readability crate

I returning `article.text` because `.text` contains the plain text content of the article, with all HTML tags removed. It is useful because I have the AI respond better to it.

We had it as `article.content`. `content` contains the cleaned HTML content of the article. It retains the HTML structure but removes unnecessary elements like ads, sidebars. However it can giet very messy if the article has pictures that model can't read through it yet. 

Maybe we can play around the AI and make it scan through images (if it sees one) 🤔.

However for now I chose to use `.text` to help the AI to output better responses! 

Happy to chat more to see your input!

Reply:
Ya love this!!

And we had pretty similar thoughts!

I was thinking that we could use the Markdown that we convert after this as the 'text version'!
My thinking is that Markdown is much closer to plain text, but can keep just a _bit_ of the formatting like titles and stuff so that the LLM can use those to help it ou!

Like I think knowing there is a heading call 'Migrations' (or whatever) is good to know over the work just being in the middle of a paragraph lol

Does that make sense? What do you think about using the markdown for that?

BUT I'm ok with using the plain text too, if you think it will be better!
But I do think I'd like to at least _store_ the cleaned html, so if we want to save the text too how about we do it in another column?

Reply:
The output of `clean_raw_html` is what is being used to store in the Markdown file. So the markdown file can be the `.text` or `.content` that is being produced at the clean_raw_html.

Reply:
No sorry!

Two suggestions take your pick between them!

- Don't store the text and pass the markdown to the LLM
- Store the txt in a new column and pass that to the LLM

Originally I was thinking we'd use the markdown, and was kinda why we did that step before this llm one!

But if you think the txt leads to better summaries I'm down to do that! Just don't want to _lose_ the cleaned html if that makes sense!

Reply:
We are already passing down the markdown to LLM and the results are looking good in my opinion. So I will go with option for now.

Worst case we can circle back and make the change and see where things go.

I think MVP goals we are fine for now. I think 🧐

Reply:
Can this go back to doing `article.content` then? So that we still save the cleaned html as opposed to just the text?

I don't want to 'lose' the HTML cause I think it will be useful later if we want to be smarter and parse things out (like the title tag!)

Reply:
> We are already passing down the markdown to LLM and the results are looking good in my opinion.

Ohhh! 🤦  I looked at the commits from this AM again and I def just read them wrong 😆 

I didn't realize the markdown was being passed to the LLM, I thought _just_ this `text` was. Sorry!

But is the cleaned html + markdown not giving as good results as the `text + markdown`?

I'd be interested in potentially also just needing in the raw 'text' at that point... Not sure if going from text -> markdown does much.. Actually does it do _anything_ when we pass the `article.text` to the markdown converter 😆 ? Is it basically the same thing back out lol?

Reply:
Yes we can go back to `article.content`. 

Just to make sure I'm explaining this correctly if we return `article.text` in the in the `clean_raw_html` function, the MD file contains just the pure texts. if we use `article.content` we return a cleaned up HTML. That includes all the `HTML tags`.

So I ran a couple of experiments.  if we want to keep the "_cleaned html_" into the MD file then we can keep it. My thesis is the AI when it scans "cleaned_html" it returns a longer response. 

If our objective is summarizing and keeping it concise then creating a `text column` in `MD table` is a good idea.

I think to figure out a the best approach, we should hop on a quick 5 min slack huddle so I can demo it and we can together take a good decision!